libs/live-config/src/schema.ts

```diff
@@ -0,0 +1,79 @@
+import { Provider } from "./providers";
+import { FirebaseRemoteConfigProvider } from "./providers/firebaseRemoteConfig";
+
+type ConfigType =
+  | {
+      type: "boolean";
+      default: true | false;
+    }
+  | {
+      type: "number";
+      default: number;
+    }
+  | {
+      type: "string";
+      default: string;
+    }
+  | {
+      type: "object";
+      default: object;
+    };
+
+export type ConfigInfo = ConfigType & {
+  provider: Provider;
+};
+
+const defaultEnabledValue = { enabled: false };
+
+export const configSchema = {
```

Comment:
why is this abstraction needed ? why do we need to specify the provider for each ? this is too heavy